entity-framework/core/miscellaneous/connection-resiliency.md

```diff
@@ -39,12 +36,57 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+An execution strategy that automatically retries on failures needs to be able to play back each operation in an retry block that fails. When retries are enabled, each operation you perform via EF Core becomes its own retriable operation, i.e. each query and each call to `SaveChanges()` will be retried as a unit if a transient failure occurs.
 
 However, if your code initiates a transaction using `BeginTransaction()` you are defining your own group of operations that need to be treated as a unit, i.e. everything inside the transaction would need to be played back shall a failure occur. You will receive an exception like the following if you attempt to do this when using an execution strategy.
 
 > InvalidOperationException: The configured execution strategy 'SqlServerRetryingExecutionStrategy' does not support user initiated transactions. Use the execution strategy returned by 'DbContext.Database.CreateExecutionStrategy()' to execute all the operations in the transaction as a retriable unit.
 
 The solution is to manually invoke the execution strategy with a delegate representing everything that needs to be executed. If a transient failure occurs, the execution strategy will invoke the delegate again.
 
-[!code-csharp[Main](../../../samples/core/Miscellaneous/ConnectionResiliency/Program.cs#Sample)]
+[!code-csharp[Main](../../../samples/core/Miscellaneous/ConnectionResiliency/Program.cs#ManualTransaction)]
+
+## Transaction commit failure and the idempotency issue
+
+In general, when there is a connection failure the current transaction is rolled back. However, if the connection is dropped while the transaction is being committed the resulting state of the transaction is unknown. See this [blog post](http://blogs.msdn.com/b/adonet/archive/2013/03/11/sql-database-connectivity-and-the-idempotency-issue.aspx) for more details.
+
+By default, the execution strategy will retry the operation as if the transaction was rolled back, but if it's not the case this will result in an exception if the new database state is incompatible or could lead to **data corruption** if the operation does not rely on a particular state, for example when inserting a new row with auto-generated key values.
+
+There are several ways to deal with this.
+
+### Option 1 - Do (almost) nothing
+
+The likelihood of a connection failure during transaction commit is low so it may be acceptable for your application to just fail if this condition actually occurs.
+
+However, you need to avoid using store-generated keys in order to ensure that an exception is thrown instead of adding a duplicate row. Consider using a client-generated GUID value or a client-side value generator such as `ForSqlServerUseSequenceHiLo`.
+
```

Comment:
We shouldn't recommend people use ForSqlServerUseSequenceHiLo. It has other implications and is confusing for people who don't understand what is happening.